### PR TITLE
fix temporary mutex and data race in NicerConnection

### PR DESCRIPTION
**Pull request by Trisfald**
**Description**

There are some mutex in ICE implementation classes that probably aren't doing their intended task, because they are instantly constructed and destructed (by the way, it seems to me these locks aren't really needed, maybe its better to remove them?).
This PR fixes this oversight and fixes also a bug in NicerConnection: there's a race between the asynchronous closeSync call and the synchronous one that is done if the former times out.

[] It needs and includes Unit Tests

**Changes in Client or Server public APIs**

[] It includes documentation for these changes in `/doc`.

### CODE REVIEW

**Comment by jcague** [comment]
wow! good catch! I don't see why we need `closeSync_mutex_` though, shouldn't we use just `close_mutex_` instead?

File: erizo/src/erizo/NicerConnection.h [hunk 1]
```diff
@@ -105,6 +105,8 @@ class NicerConnection : public IceConnection, public std::enable_shared_from_thi
   nr_ice_handler* ice_handler_;
   std::promise<void> close_promise_;
   std::promise<void> start_promise_;
+  boost::mutex close_mutex_;
+  boost::mutex closeSync_mutex_;
```
Comment (jcague) [inline erizo/src/erizo/NicerConnection.h]: please consider using snake_case here: `close_sync_mutex_`

**Comment by Trisfald** [comment]
Glad to be of help!

File: erizo/src/erizo/NicerConnection.h (hunk 1 above)
Comment (jcague) [reply erizo/src/erizo/NicerConnection.h]: @Trisfald The only thing we need to merge this PR is this change, could you apply it please? thanks a lot!

Comment (Trisfald) [reply erizo/src/erizo/NicerConnection.h]: Done! By bad, I don't know how I missed your prior comment

**Pull request merged**